adagrams/game.py

```diff
@@ -1,11 +1,156 @@
-    pass
+    """
+    Create a dictionary looping over letter in letter_bank 
+    Create key using letter, sum quantities of given letter key
+    Create dictionary for letter in word, similarly 
+    Returns True or False
+    """
+    cap_word = word.upper()
+    letter_bank_quantity = {}
+    char_quantity = {}
+    result = None #flag called result 
+
+    for letter in letter_bank:
+        if letter not in letter_bank_quantity:
+            letter_bank_quantity[letter] = 1
+        else:
+            letter_bank_quantity[letter] += 1
+    
+    for char in cap_word:
+        if char not in char_quantity:
+            char_quantity[char] = 1
+        else:
+            char_quantity[char] += 1
+    
+    for letter in cap_word:
+        if letter not in letter_bank or char_quantity[letter] > letter_bank_quantity[letter]:
+            result = False 
+        else:
+            result = True
+
+    return result  
+
 
 def score_word(word):
-    pass
+    """
+    Define a list of tuples with group of letter's point value
+    Loop over letter_score list to find if each character in word is in list
+    If char is found in that index, score will be same index of outer list, index 1 of inner list
+    Returns sum_score, holds total points for given word 
+    """
+    letter_score = [
+        [("A", "E", "I", "O", "U", "L", "N", "R", "S", "T"), 1],
+        [("D", "G"), 2],
+        [("B", "C", "M", "P"), 3],
+        [("F", "H", "V", "W", "Y"), 4],
+        [("K"), 5],
+        [("J", "X"), 8],
+        [("Q", "Z"), 10]
+    ]
+    
+    cap_word = word.upper()
+    sum_score = 0
+
+    for char in cap_word:
+        for i in range(len(letter_score)):
+            if char in letter_score[i][0]: #check if char is in list of letters in inner list index 0
+                sum_score += letter_score[i][1] #sums integer in inner list index 1
+    
+    if len(word) >= 7:
+        sum_score += 8
+
+    return sum_score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    Create a list of tuples containing word from word_list and score 
+    Calculate score calling score_word function for word
+    Each word and corresponding score will be at same index of list 
+    Returns tuple of highest scoring word and corresponding score
+    """
+    words_and_score = []
+    
+    index_of_highest_tuple = 0 #keeps track of index where the current highest scoring word is located
+    highest_score = 0 
+
+    for word in word_list:
+        score = score_word(word)
+        word_tuple = (word, score)
+        words_and_score.append(word_tuple)
+    
+    for i in range(len(words_and_score)):
+        score = words_and_score[i][1]
+        word = words_and_score[i][0]
+        if score > highest_score:
+            highest_score = score
+            index_of_highest_tuple = i #assigns index of word currently being checked in for loop
+        elif score == highest_score:
```

Comment:
In my opinion, adding a comment describing the specific contextual rules of the tie-breaking could add clarity. (I also truly believe it's fine without)